[[ BUG 8181 ]] Move a File to Trash or Recycle Bin

lan-kc opened the pull request:
Added a sentence to the end of the Warniing + the accompanying example.

NOTE: I've only tested the code for OS X as I do not have a Windows machine. The VB script was extracted from the Use List and dates back to 2008 so verification that it is still valid is required. Sorry, my research for Linux indicated that the Trash location changed with distribution and/or windowing environment so no Linux example is given.

Bug 8181 refers to a Caution for both delete file and delete folder but no such Caution (now Warning) exists in the current documents for delete folder - nor should one exist as you can only delete an empty folder with this command. I therefore see no need to add an example of how to move an empty folder to the Trash or Recycle Bin. I intend to add an example of how to move a 'full' folder to the Trash for the revDeleteFolder command; at which point bug 8181 should be closed.

lan-kc committed:
[[8181]] Move a File to Trash or Recycle Bin

Added a sentence to the end of the Warniing + the accompanying example.

NOTE: I've only tested the code for OS X as I do not have a Windows machine. The VB script was extracted from the Use List and dates back to 2008 so verification that it is still valid is required. Sorry, my research for Linux indicated that the Trash location changed with distribution and/or windowing environment so no Linux example is given.

Bug 8181 refers to a Caution for both delete file and delete folder but no such Caution (now Warning) exists in the current documents for delete folder - nor should one exist as you can only delete an empty folder with this command. I therefore see no need to add an example of how to move an empty folder to the Trash or Recycle Bin. I intend to add an example of how to move a 'full' folder to the Trash for the revDeleteFolder command; at which point bug 8181 should be closed.

lan-kc commented:
As usual a split second after clicking the Create Pull Request button I noticed an error.

This line:
--on OS X the Trash is ~/.Trash/ so all you need to do is rename your file to '/.Trash/fileToTrash.
Should inlcude the ~ in the second instance
--on OS X the Trash is ~/.Trash/ so all you need to do is rename your file to '~/.Trash/fileToTrash.

As this was my first Pull Request I couldn't figure out how to quickly go back and amend it - but I'm still looking.

lan-kc committed:
typo fixed

lan-kc commented:
OK, figured out how to go back and fix the typo.

peter-b commented:
@lan-kc Did you want to merge these changes up into the main LiveCode repository?  I think you've maybe opened a PR against your own personal fork, rather than against our upstream repo!

/cc @livecodeali

lan-kc commented:
Hi Peter,

Yes thanks. As this was my first attempt at using Git I think I took the
too cautious approach. Then I thought I'd got it, and then... maybe not -
you'll find another one for BUG 15176. But I figured I'd eventually get a
gentle prod in the right direction. :-)

Slowly figuring it out.... I think.

On Mon, Jan 25, 2016 at 6:13 PM, Peter TB Brett notifications@github.com
wrote:

> @lan-kc https://github.com/lan-kc Did you want to merge these changes
> up into the main LiveCode repository? I think you've maybe opened a PR
> against your own personal fork, rather than against our upstream repo!
> 
> /cc @livecodeali https://github.com/livecodeali
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/lan-kc/livecode/pull/1#issuecomment-174460197.

livecodeali commented:
Hi @lan-kc , thanks very much for this!

`Example:` is used specifically to add an example element to the docs. In this case, I think you have the code example in the correct place (as it is not specifically an example of using _delete file_, so should be in the `Description:` element). 

As such you should use normal markdown to render it as a code block within the description element, instead of using `Example:`. The easiest way to do this would be to tab-indent the code block you have added.

I would suggest adding the rename command to the references, and linking to it in the text that introduces the code, i.e. something like:

```
To place a file in the Trash or Recycle Bin use the <rename> command, as in the following example:
```

Finally, could you add a file called docs/notes/bugfix-8181.md with the contents 

```
# Entry for delete file should describe how to move file into trash / recycle bin
```

This is for our release notes.

To open a new pull request to our repo, you can go [here](https://github.com/lan-kc/livecode/tree/lan-kc-patch-1) and click "New Pull Request", and make sure the base fork is livecode/livecode.

lan-kc commented:
Thanks Ali,

Yes, part of my struggle is that as far as I can tell there is no way to
'Preview' the changes to see how they would look in the Dictionary.

I have already added a bugfix-8181.md file (and also bugfix-15176.md) but
if you can't find it maybe I added it in the wrong place.

I am trying to follow the instructions here:

https://github.com/livecode/livecode/blob/community-docs/docs/contributing_to_docs.md#using-the-github-web-interface

but one of the problems I'm having is that steps like this:

---

Check the pull request again in https://github.com/livecode/livecode/pulls

## You should see the "Add release note" commit added to the pull request.

don't work for me. To get any where I always seem to have my 'username' in
the URL:

https://github.com/lan-kc/livecode/pulls

So treading carefully but hoping I'll eventually get there.

On Mon, Jan 25, 2016 at 8:08 PM, Ali Lloyd notifications@github.com wrote:

> Hi @lan-kc https://github.com/lan-kc , thanks very much for this!
> 
> Example: is used specifically to add an example element to the docs. In
> this case, I think you have the code example in the correct place (as it is
> not specifically an example of using _delete file_, so should be in the
> Description: element).
> 
> As such you should use normal markdown to render it as a code block within
> the description element, instead of using Example:. The easiest way to do
> this would be to tab-indent the code block you have added.
> 
> I would suggest adding the rename command to the references, and linking
> to it in the text that introduces the code, i.e. something like:
> 
> To place a file in the Trash or Recycle Bin use the <rename> command, as in the following example:
> 
> Finally, could you add a file called docs/notes/bugfix-8181.md with the
> contents
> 
> # Entry for delete file should describe how to move file into trash / recycle bin
> 
> This is for our release notes.
> 
> To open a new pull request to our repo, you can go here
> https://github.com/lan-kc/livecode/tree/lan-kc-patch-1 and click "New
> Pull Request", and make sure the base fork is livecode/livecode.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/lan-kc/livecode/pull/1#issuecomment-174486837.

livecodeali commented:
@lan-kc Ok, thanks for that - it is good to get feedback on the instructions so I can make them as useful as possible!

The reason the pull request doesn't appear in https://github.com/livecode/livecode/pulls is that the original was targeted to your fork rather than the livecode fork - once you have created one that targets the livecode fork it will appear in that list.

lan-kc commented:
Answering two emails here:

Firstly, yes I appreciate now that my too cautious approach sent me up the
garden path - or in this case my own Branch. I think I may have figured
that out now. Probably should have stopped at one Doc amendment and Note
addition and waited for these comments.

Secondly, as per my previous statement about not having a 'Preview' it was
in the 2nd attempt with Bug 15176 that I 'saw' the < > tags and started to
appreciate their worth - this particular entry seemed to be very well
linked with copious examples.

I fully agree with your comment that a simple link to 'rename' would be
much tidier in both cases. If you haven't already done it I'd be happy to
give it a go - but it will have to wait until tomorrow as it's bed time
here.

Let me know.

On Mon, Jan 25, 2016 at 8:34 PM, Ali Lloyd notifications@github.com wrote:

> @lan-kc https://github.com/lan-kc Ok, thanks for that - it is good to
> get feedback on the instructions so I can make them as useful as possible!
> 
> The reason the pull request doesn't appear in
> https://github.com/livecode/livecode/pulls is that the original was
> targeted to your fork rather than the livecode fork - once you have created
> one that targets the livecode fork it will appear in that list.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/lan-kc/livecode/pull/1#issuecomment-174492416.
